AK/Atomic.h

```diff
@@ -279,7 +279,13 @@ class Atomic<T, DefaultMemoryOrder> {
 
     ALWAYS_INLINE T fetch_sub(T val, MemoryOrder order = DefaultMemoryOrder) volatile noexcept
     {
-        return __atomic_fetch_sub(&m_value, val, order);
+        T volatile* ptr = &m_value;
+        // FIXME: GCC > 12 will wrongly warn on -Wstringop-overflow here with ASAN+UBSAN
+#if defined(AK_COMPILER_GCC)
```

Comment:
Can we make it `defined(AK_COMPILER_GCC) && defined(__SANITIZE_ADDRESS__)`?

Reply:
sure, but I'll use HAS_ADDRESS_SANITIZER from AK/Platform.h:235

Reply:
GCC PR # for this?

Reply:
Not sure... did we have one to reference before? I think someone found one where the gcc devs said :shrug: sanitizers have false positives